x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1154,13 +1154,33 @@ public CacheFile getCacheFile(KeyType cacheKey, long length) {
 
     @FunctionalInterface
     public interface RangeAvailableHandler {
-        // caller that wants to read from x should instead do a positional read from x + relativePos
-        // caller should also only read up to length, further bytes will be offered by another call to this method
+        /**
+         * A strategy used to read the blob from the local (cached) blob. The target is typically captured by the strategy implementation.
+         *
+         * A caller that wants to read from position p should instead do a positional read from p + relativePos
+         * A caller should also only read up to length, further bytes will be offered by another call to this method
+         *
+         * @param channel is the cache region to read from
+         * @param channelPos a position in the channel to read from
+         * @param relativePos
```

Comment:
Should we document this?

Reply:
Sure, could you please confirm that here it is also a position in the remote storage (similarly to RangeMissingHandler)?
I also wonder if could be renamed to something else to make it easier to understand. May be sourcePos?

Reply:
I think this is the relative position in the destination, not in the remote storage (source). This is the position from the caller point of view where the range to read starts.

Reply:
I find the two `relativePos` parameters to be the most confusing ones in the interfaces. What are they relative to exactly? I also don't mind if they are named differently since they are for different things.